test/unit/storage_async_test.rb

```diff
@@ -182,20 +190,98 @@ def test_sentinels_empty
+  def test_tls_no_client_certificate
+    config_obj = {
+      url: 'rediss://localhost:46379/0',
+      ssl_params: {
+        ca_file: File.expand_path(File.join(__FILE__, '..', '..', '..', 'script', 'config', 'ca-root-cert.pem'))
+      }
+    }
+    storage = StorageAsync::Client.send :new, Storage::Helpers.config_with(config_obj)
+    assert_client_config(config_obj, storage)
+  end
+
+  [:rsa, :dsa, :ec].each do |alg|
+    define_method "test_tls_client_cert_#{alg}" do
+      ca_file = create_cert
+      key = create_key alg
+      cert = create_cert key
+      FakeFS.with_fresh do
+        ca_file_path = File.expand_path(File.join(__FILE__, '..', '..', '..', 'script', 'config', 'ca-root-cert.pem'))
+        key_path = File.expand_path(File.join(__FILE__, '..', '..', '..', 'script', 'config', 'redis-dsa.pem'))
+        cert_path = File.expand_path(File.join(__FILE__, '..', '..', '..', 'script', 'config', 'redis-dsa.crt'))
+        FileUtils.mkdir_p(File.dirname(ca_file_path))
+        File.write(ca_file_path, ca_file.to_pem)
+        File.write(key_path, key.to_pem)
+        File.write(cert_path, cert.to_pem)
+        config_obj = {
+          url: 'rediss://localhost:46379/0',
+          ssl_params: {
+            ca_file: ca_file_path,
+            cert: cert_path,
+            key: key_path
+          }
+        }
+        storage = StorageAsync::Client.send :new, Storage::Helpers.config_with(config_obj)
+        assert_client_config(config_obj, storage, alg)
+      end
+    end
+  end
+
+  def test_acl
+    config_obj = {
+      url: 'redis://localhost:6379/0',
+      username: 'apisonator-test',
+      password: 'p4ssW0rd'
+    }
+    storage = StorageAsync::Client.send :new, Storage::Helpers.config_with(config_obj)
+    assert_client_config(config_obj, storage)
+  end
+
+  def test_acl_tls
+    config_obj = {
+      url: 'rediss://localhost:46379/0',
+      ssl_params: {
+        ca_file: File.expand_path(File.join(__FILE__, '..', '..', '..', 'script', 'config', 'ca-root-cert.pem'))
+      },
+      username: 'apisonator-test',
+      password: 'p4ssW0rd'
+    }
+    storage = StorageAsync::Client.send :new, Storage::Helpers.config_with(config_obj)
+    assert_client_config(config_obj, storage)
   end
 
   private
 
-  def assert_client_config(conf, conn)
+  def assert_client_config(conf, conn, test_cert_type = nil)
     client = conn.instance_variable_get(:@redis_async)
 
     url = URI(conf[:url])
     host, port = client.endpoint.address
     assert_equal url.host, host
     assert_equal url.port, port
+
+    unless conf[:username].to_s.strip.empty? && conf[:password].to_s.strip.empty?
+      assert_instance_of ThreeScale::Backend::AsyncRedis::Protocol::ExtendedRESP2, client.protocol
+      username, password = client.protocol.instance_variable_get(:@credentials)
+      assert_equal conf[:username], username
+      assert_equal conf[:password], password
```

Comment:
I notice no tests for sentinels credentials. Maybe add one?

Reply:
Done